lib/util.js

```diff
@@ -478,7 +478,22 @@ function formatValue(ctx, value, recurseTimes) {
   if (ctx.showHidden) {
     keys = Object.getOwnPropertyNames(value);
   } else {
-    keys = Object.keys(value);
+    // This might throw if `value` is a Module Namespace Object from an
+    // unevaluated module, but we don't want to perform the actual type
+    // check because it's expensive.
+    // TODO(devsnek): track https://github.com/tc39/ecma262/issues/1209
+    // and modify this logic as needed.
+    try {
```

Comment:
* `Object.keys` and `Object.getOwnPropertyNames` references should be cherry-picked above.
* The `instanceof` ReferenceError check will fail from errors of a different realm (context).

Reply:
for the first point, we purposely leave those as-is because some polyfills for things use them to mess with util.

for the second point, i guess `types.isNativeError(err) && err.name === 'ReferenceError'` would work?

Reply:
> for the second point, i guess types.isNativeError(err) && err.name === 'ReferenceError' would work?

Yep.

Reply:
> for the first point, we purposely leave those as-is because some polyfills for things use them to mess with util.

Ah cool! Do you have any examples at hand?

Reply:
I don't know what package does it, but it was brought up during discussions about hardening core against tampering with builtin prototypes.